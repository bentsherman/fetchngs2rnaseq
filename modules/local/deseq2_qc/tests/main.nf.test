nextflow_process {

    name "Test Process DESEQ2_QC"
    script "../main.nf"
    process "DESEQ2_QC"

    test("parse count data correctly") {

        when {
            process {
                """
                input[0] = file(params.pipelines_testdata_base_path + "deseq2qc/countFile.tsv", checkIfExists: true)
                input[1] = file(params.pipelines_testdata_base_path + "deseq2qc/deseq2_pca_header.txt", checkIfExists: true)
                input[2] = file(params.pipelines_testdata_base_path + "deseq2qc/deseq2_clustering_header.txt", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.dists_multiqc,
                    process.out.dists_txt,
                    process.out.log,
                    process.out.pca_multiqc,
                    process.out.pca_txt,
                    process.out.rdata,
                    process.out.size_factors
                ).match() }
            )
        }

    }

}
