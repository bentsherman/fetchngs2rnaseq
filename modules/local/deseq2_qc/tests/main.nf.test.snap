{
    "parse count data correctly": {
        "content": [
            [
                
            ],
            [
                "deseq2.sample.dists.txt:md5,1616ce40eb58ee75640efed219c6bbec"
            ],
            [
                "R_sessionInfo.log:md5,fb0da0d7ad6994ed66a8e68348b19676"
            ],
            [
                
            ],
            [
                "deseq2.pca.vals.txt:md5,4cda8e32ec39326c1350e77b4b607aad"
            ],
            [
                "deseq2.dds.RData:md5,c7c1872e7566321d0c3f7ca56c463088"
            ],
            [
                [
                    "RAP1_IAA_30M_REP1.txt:md5,ce4ff0b74aeedc3dc12a764c3504c9df",
                    "RAP1_UNINDUCED_REP1.txt:md5,51c9b23be5e82670aceb14f7e58dd867",
                    "RAP1_UNINDUCED_REP2.txt:md5,2c56106abcd9d31980f672b6e2862ee8",
                    "WT_REP1.txt:md5,5e3fe5f7e9b7f300be43649a8b496352",
                    "WT_REP2.txt:md5,2ce093f2c2e92c1bde1cd3da91ac884c",
                    "deseq2.size_factors.RData:md5,51c6643a0f8a9edabbff8df0165451f2"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-07T12:34:51.582179"
    }
}